build.gradle.kts

```diff
@@ -2,10 +2,42 @@ plugins {
     //trick: for the same plugin versions in all sub-modules
     id("com.android.application").version("8.0.2").apply(false)
     id("com.android.library").version("8.0.2").apply(false)
+    id("io.gitlab.arturbosch.detekt").version("1.23.0")
     kotlin("android").version("1.8.21").apply(false)
     kotlin("multiplatform").version("1.8.21").apply(false)
 }
 
+detekt {
+    toolVersion = "1.23.0"
+    config.setFrom("detekt.yml")
+
+    source.setFrom(
+        "android/src",
+        "shared/src/androidMain/kotlin",
+        "shared/src/androidUnitTest/kotlin",
+        "shared/src/commonMain/kotlin",
+        "shared/src/commonTest/kotlin",
+        "shared/src/iosMain/kotlin",
```

Comment:
@lydiasama What is the issue when running detekt on `shared/src/iosTest`?

Reply:
@lydiasama Detekt fails because there are 11 issues (> 10) after adding iosTest lol

![image](https://github.com/nimblehq/kmm-templates/assets/16315358/82e39260-a185-464e-a809-ae5f6a36eb2d)

Increasing the `maxIssues` threshold will make it pass. However, 10 is the ideal number. I bet you will need to resolve some of them in Part 2 with the CI integrated 😂 

![image](https://github.com/nimblehq/kmm-templates/assets/16315358/2f288701-cde4-4f20-9605-03949f835091)

Reply:
So it's not about the tooling (we can't run detekt on iosTests), but it's more about the code does not pass the detekt rules. If it's the latter, then we need to fix the code 🥹

Reply:
@suho @lydiasama This PR is to set up the Detekt, we can resolve its issues to pass the CI in Part 2 🙏 

@lydiasama Please also correct the PR title to `Set up Detekt` as no CI workflow is set up yet 🙏